### PR TITLE
Added newer models for split learning

### PR DESCRIPTION
**Pull request by rsk97**
Added ResNet-34 and ResNet-50.

### CODE REVIEW

File: src/algos/simba_algo.py [hunk 1]
```diff
@@ -33,6 +33,12 @@ def init_client_model(self, config):
         if config["model_name"] == "resnet18":
             model = models.resnet18(pretrained=config["pretrained"])
             model = nn.Sequential(*nn.ModuleList(list(model.children())[:config["split_layer"]]))
+        elif config["model_name"] == "resnet34":
```
Comment (tremblerz) [inline src/algos/simba_algo.py:35]: Since the second line is same for all of them, can you create the model inside `if` and assign this model `if config["model_name"].startswith("resnet")`

File: src/configs/split_infer_cifar.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (tremblerz) [inline src/configs/split_infer_cifar.json:1]: is it working?

File: src/configs/split_infer_res34.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (tremblerz) [inline src/configs/split_infer_res34.json:1]: Instead of creating new configs for each model, let's have a README section that talks about different arguments supported for the key `model_name`

File: src/models/model_zoo.py [hunk 4]
```diff
@@ -45,6 +45,26 @@ def assign_model(self, config):
                                         nn.Linear(num_ftrs, logits))
                 model = nn.ModuleList(list(model.children())[self.split_layer:])
                 self.model = nn.Sequential(*model)
+            elif config["model_name"] == "resnet34":
+                pretrained = config["pretrained"]
```
Comment (tremblerz) [inline src/models/model_zoo.py:45]: Same comment as the client model to minimize the redundant code

File: src/utils/logs.py [hunk 5]
```diff
@@ -15,7 +15,8 @@ def __init__(self, config, exp_type):
         self.curr_iters = 0
         self.total_epochs = config["total_epochs"]
         self.items = {}
-        self.metrics = MetricLoader()
+        self.metrics = MetricLoader()
```
Comment (tremblerz) [inline src/utils/logs.py:18]: Can you fix the code inside metricloader to work with cpu?

**Review by tremblerz — CHANGES_REQUESTED** [review]

File: src/configs/split_infer_cifar.json (hunk 2 above)
Comment (rsk97) [reply src/configs/split_infer_cifar.json:1]: Yeah, this is the old CIFAR code. I did run it and saw loss go down (only ran for few epochs though)

File: src/configs/split_infer_res34.json (hunk 3 above)
Comment (rsk97) [reply src/configs/split_infer_res34.json:1]: Yeah, makes sense. I will update the readme. Also, do you want me to remove these config files (I had made them for testing purposes)

File: src/utils/logs.py (hunk 5 above)
Comment (rsk97) [reply src/utils/logs.py:18]: Sure, for me passing the device argument as "CPU" worked. I have commented that in line 19. Are you suggesting changing things inside the metric loader so that the onus isn't on the user to pass this argument?

File: src/algos/simba_algo.py (hunk 1 above)
Comment (rsk97) [reply src/algos/simba_algo.py:35]: Yeah, will update this

**Comment by rsk97** [comment]
Removed the new config files

File: src/utils/logs.py [hunk 6]
```diff
@@ -15,7 +15,8 @@ def __init__(self, config, exp_type):
         self.curr_iters = 0
         self.total_epochs = config["total_epochs"]
         self.items = {}
-        self.metrics = MetricLoader()
+        self.metrics = MetricLoader() 
+        # self.metrics = MetricLoader(device = "cpu") # need to pass device as CPU to not throw errors in non-CUDA machines
```
Comment (tremblerz) [inline src/utils/logs.py:19]: Metricloader should figure out on its own whether the device is cpu

Comment (rsk97) [reply src/utils/logs.py:19]: Cool! I'll fix that!
